force-app/main/default/lwc/datatableMultilineEdit/__tests__/datatableMultilineEdit.test.js

```diff
@@ -0,0 +1,259 @@
+
+    it('updates a record on Save', async () => {
+        const INPUT_PARAMETERS = [{ data: DRAFT_VALUES }];
+
+        // Create initial element
+        const element = createElement('c-datatable-multiline-edit', {
+            is: DatatableMultilineEdit
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for save event
+        const handler = jest.fn();
+        // Add event listener to catch save event
+        element.addEventListener('save', handler);
+
+        // Emit data from @wire
+        getContacts.emit(mockGetContactList);
+
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        const tableEl = element.shadowRoot.querySelector('lightning-datatable');
+        tableEl.dispatchEvent(
+            new CustomEvent('save', {
+                detail: {
+                    draftValues: DRAFT_VALUES
+                }
+            })
+        );
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        // Validate updateContacts call
+        expect(updateContacts).toHaveBeenCalled();
+        //Validate the update call is made for only one record
+        expect(updateContacts.mock.calls[0]).toEqual(INPUT_PARAMETERS);
+    });
+
+    it('displays a success toast after record is updated', async () => {
+        //Update all the records in the Draft Values
+        const INPUT_PARAMETERS = [{ data: DRAFT_VALUES }];
+
+        // Assign mock value for resolved updateContacts promise
+        updateContacts.mockResolvedValue(INPUT_PARAMETERS);
+
+        // Create initial element
+        const element = createElement('c-datatable-multiline-edit', {
+            is: DatatableMultilineEdit
+        });
+        document.body.appendChild(element);
+
+        const handler = jest.fn();
+        element.addEventListener('save', handler);
+
+        // Mock handler for toast event
+        const toastHandler = jest.fn();
+        // Add event listener to catch toast event
+        element.addEventListener(ShowToastEventName, toastHandler);
+
+        // Emit data from @wire
+        getContacts.emit(mockGetContactList);
+
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        const tableEl = element.shadowRoot.querySelector('lightning-datatable');
+        tableEl.dispatchEvent(
+            new CustomEvent('save', {
+                detail: {
+                    draftValues: DRAFT_VALUES
+                }
+            })
+        );
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        expect(toastHandler).toHaveBeenCalled();
+        expect(toastHandler.mock.calls[0][0].detail.variant).toBe('success');
+        //Validate refreshApex is called and the draft values are reset
+        expect(refreshApex).toHaveBeenCalled();
+        expect(tableEl.draftValues).toEqual([]);
+    });
+
+    it('displays an error toast on update record error', async () => {
```

Comment:
Change this test so that we use the error panel (best practice) to convey errors instead of a toast

Reply:
The user will be modifying one or two records. The error panel replacing the table or below the table might look weird. Added toast messages similar to Data Service recipes. I can change it if required :)

Reply:
Ok, let's leave it as is for now.